src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java

```diff
@@ -102,9 +105,9 @@ private void processAction(Action action) {
       }
 
       if (action.getState().equals(ActionState.SUBMITTED)) {
-        ap.processActionRequests(action);
+        ap.processActionRequests(action.getId());
```

Comment:
**absolutetely**

The Action which was returned in the Stream is an entity which is attached to one particular transaction/Hibernate session, but we then explicitly start _another_ transaction, which updates some of the very same data which the _first_ transaction returned in its find query.

The net result - without re-getting the same Action from the DB again - would be that the new transaction can commit without any problems, but the original transaction will log an error, because its data was meddled with.

Basically, we shouldn't pass Hibernate/JPA persistent entities around in our code, crossing transaction boundaries, because it then makes it very unclear which Hibernate session and which transaction is responsible for committing any changes.